### PR TITLE
[MySQL] Moving common mysql utility to ambry-commons module

### PR DESCRIPTION
**Pull request by justinlin-linkedin**
Moving several MySql classes from accounts to commons module.

### CODE REVIEW

**Review by cgtz — CHANGES_REQUESTED** [review]
Changes look good but there seems to be a compilation error on travis

I was initially wondering if it would be good to create a separate ambry-mysql subproject to separate out the mysql dependencies, but the mysql connector is not a huge dependency and is probably needed by most of the subprojects anyways, so commons should be fine.

**Comment by justinlin-linkedin** [comment]
> Changes look good but there seems to be a compilation error on travis
> 
> I was initially wondering if it would be good to create a separate ambry-mysql subproject to separate out the mysql dependencies, but the mysql connector is not a huge dependency and is probably needed by most of the subprojects anyways, so commons should be fine.

The compilation error is already fixed.

**Review by cgtz — APPROVED** [review]
lgtm

File: build.gradle [hunk 1]
```diff
@@ -230,6 +230,7 @@ project(':ambry-commons') {
         compile ("org.apache.helix:helix-core:$helixVersion") {
             exclude group: "log4j", module: "log4j"
         }
+        compile "mysql:mysql-connector-java:8.0.21"
```
Comment (lightningrob) [inline build.gradle:233]: Could you add $mysqlVersion and use it here.  Also, can we remove it from account project?

File: ambry-commons/src/main/java/com/github/ambry/mysql/MySqlMetrics.java [hunk 2]
```diff
@@ -11,17 +11,17 @@
...
 import com.codahale.metrics.Histogram;
 import com.codahale.metrics.MetricRegistry;
 
 
 /**
- * Expose metrics related to writes and reads to/from MySql DB used in {@link com.github.ambry.account.MySqlAccountService}
+ * Expose metrics related to writes and reads to/from MySql DB.
  */
-public class MySqlAccountStoreMetrics {
+public class MySqlMetrics {
```
Comment (lightningrob) [inline ambry-commons/src/main/java/com/github/ambry/mysql/MySqlMetrics.java:24]: Good idea, this is independent of accounts.

File: ambry-commons/src/main/java/com/github/ambry/mysql/MySqlDataAccessor.java [hunk 3]
```diff
@@ -317,7 +286,7 @@ private boolean isBetterEndpoint(DbEndpoint first, DbEndpoint second) {
    * Close a resource without throwing exception.
    * @param resource the resource to close.
    */
-  private static void closeQuietly(AutoCloseable resource) {
+  private void closeQuietly(AutoCloseable resource) {
```
Comment (lightningrob) [inline ambry-commons/src/main/java/com/github/ambry/mysql/MySqlDataAccessor.java:289]: Why not static?  (This can probably move to MySqlUtils anyway.)

**Review by lightningrob — APPROVED** [review]
LGTM.  A few minor comments.

**Review by Arun-LinkedIn — APPROVED** [review]
Looks good to me.

File: build.gradle (hunk 1 above)
Comment (justinlin-linkedin) [reply build.gradle]: updated

File: ambry-commons/src/main/java/com/github/ambry/mysql/MySqlDataAccessor.java (hunk 3 above)
Comment (justinlin-linkedin) [reply ambry-commons/src/main/java/com/github/ambry/mysql/MySqlDataAccessor.java:289]: I can revert the change.

**Pull request merged**